cmake/modules/CMakeFunctions.cmake

```diff
@@ -97,19 +97,20 @@ function(apply_patches repo_dir patches_dir base_revision target_branch ret)
         return()
     endif()
 
-    message(STATUS "[OPENCL-CLANG] Patching repository ${repo_dir}")
+    # Check if it's a git repo
+    if(EXISTS "${repo_dir}/.git")
```

Comment:
can command in https://stackoverflow.com/questions/2044574/determine-if-directory-is-under-git-control/2044714#comment36357325_2044714 be used to detect if a folder is a git repo?

Reply:
it seems `git rev-parse --is-inside-work-tree` can't work. I copy opencl-clang source code to /export/users/haonanya/ocl-clang/llvm/projects/, and git rev-parse --is-inside-work-tree return true.

[haonanya@shliclel4059 opencl-clang]$ git rev-parse --is-inside-work-tree
true

Reply:
I see